R/utility.R

```diff
@@ -204,3 +204,13 @@ isUrl <- function(x) {
   regex <- "^https?://.+"
   grepl(regex, x)
 }
+
+isUrlOrPath <- function(x) {
+  stopifnot(
+    is.character(x)
+  )
+  if (!isUrl(x) && !file.exists(x)) {
```

Comment:
Oops. There were a few more changes required to "vectorize" this function. The below should properly evaluate a character vector and only evaluate to `TRUE` when there is at least one invalid element:

```suggestion
  if (any(!isUrl(x) & !file.exists(x))) {
```